languageserver/integration/integration.go

```diff
@@ -111,14 +279,68 @@ func (i *FlowIntegration) initialize(initializationOptions any) error {
 		return nil
 	}
 
-	// Load the config state if provided
+	// Load the config state if provided. If it fails, don't fail initialization;
+	// still set the init-config override so the server can run and the user can fix the file.
 	configPath = cleanWindowsPath(configPath)
 	err := i.state.Load(configPath)
 	if err != nil {
-		return err
+		// Seed cfgManager with the init-config path so project scoping still works
+		if i.cfgManager != nil {
```

Comment:
The existing function is already very large, and the added code makes it very complex. Maybe try breaking out the added parts into separate functions